pkg-manager/resolve-dependencies/src/index.ts

```
@@ -319,10 +321,89 @@ function verifyPatches (
+  }
+}
+
+async function fetchPkgIndex (pkgName: string, pkgVersion: string): Promise<RegistryFileList> {
+  const url = `https://npmjs.com/package/${pkgName}/v/${pkgVersion}/index`
+  const fetchResult = await fetch(url)
+  if (!fetchResult.ok) {
+    throw new PnpmError('PKG_INDEX_FETCH', `Failed to fetch ${pkgName}@${pkgVersion} Status: ${fetchResult.statusText}`)
+  }
+
+  const fetchJSON = await fetchResult.json() as RegistryFileList
+  if (fetchJSON.files == null) {
+    throw new PnpmError('PKG_INDEX_FILE_PARSE', `Unable to parse file list for ${pkgName}@${pkgVersion} Status: ${fetchResult.statusText}`)
+  }
+  return fetchJSON
+}
+
+async function fetchSpecificFileFromRegistryFS (pkgName: string, pkgVersion: string, fileHex: string): Promise<string> {
+  const url = `https://npmjs.com/package/${pkgName}/file/${fileHex}`
+  const fetchResult = await fetch(url)
+  if (!fetchResult.ok) {
+    throw new PnpmError('REG_FS_ERROR', `Failed to fetch ${pkgName}@${pkgVersion}, file: ${fileHex} Status: ${fetchResult.statusText}`)
+  }
+  return fetchResult.text()
+}
+
+function pkgJsonFromRegFSHasInstallScripts (pkgJson: ProjectManifest): boolean {
+  if (!pkgJson.scripts) return false
+  return Boolean(pkgJson.scripts.preinstall) ||
+    Boolean(pkgJson.scripts.install) ||
+    Boolean(pkgJson.scripts.postinstall)
+}
+
+async function fetchBuildFromRegistryFS (pkgName: string, pkgVersion: string, pkgUrl: string, useExperimentalNpmjsFilesIndex: boolean | string[]): Promise<boolean> {
+  let registryUrls = ['registry.npmjs.org/', 'registry.npmjs.com/']
+  if (Array.isArray(useExperimentalNpmjsFilesIndex)) {
+    registryUrls = useExperimentalNpmjsFilesIndex
+  }
+  // check if any of the registryUrls are contained in the pkgUrl
+  const registryUrl = registryUrls.find((url) => pkgUrl.startsWith(url))
+  if (!registryUrl) {
+    globalWarn(`${pkgName}@${pkgVersion} (${pkgUrl}) is not in NPMJS Files Index URL Allow List: Fallback to requiresBuild=true`)
```

Don't do this fallback. This will mess up the lockfile.

Just let it fail.